myning/utils/species_rarity.py

```diff
@@ -38,7 +40,23 @@
 SPECIES_WEIGHTS = [150, 75, 40, 20, 10, 7, 4]
 
 
-def get_recruit_species(highest_rarity: int):
+def _recruit_in_tier(tier: list[CharacterSpecies]) -> Optional[Species]:
+    player = Player()
+
+    def is_undiscovered(species_name: CharacterSpecies):
+        return SPECIES[species_name] not in player.discovered_species
```

Comment:
could be defined outside this function, although you would need to add another `player = Player()`

Reply:
What do you think of just a named lambda function?

Reply:
well the point is that it's not necessary to redefine this function on each call of the outer function, so might as well just make a regular function